src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  green = 'green',
+  blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const p: number = (a + b + c) / 2;
+    const sTriangle: number = Math.sqrt(p * (p - a) * (p - b) * (p - c));
 
+    return Number(sTriangle.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.Circle;
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const pi = Math.PI;
+    const area = pi * (this.radius ** 2);
```

Comment:
The variable for PI is redundant here:
```suggestion
    const area = Math.PI * (this.radius ** 2);
```